### PR TITLE
Add transform to fix Safari 10/11 block shadowing

### PR DESCRIPTION
**Pull request by developit**
To be replicated in @nicolo-ribaudo's PR.

### CODE REVIEW

File: src/plugins/transform-safari-block-shadowing/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ */
+export default function({ types: t }) {
+  return {
+    name: "transform-safari-block-shadowing",
+    visitor: {
+      VariableDeclarator(path) {
+        const kind = path.parent.kind;
+        if (kind !== "let" && kind !== "const") return;
+
+        const name = path.node.id.name;
```
Comment (nicolo-ribaudo) [inline src/plugins/transform-safari-block-shadowing/index.js]: Here you need to call `t.getOuterBindingIdentifiers(path.node.id)`, for cases like this: https://astexplorer.net/#/gist/f39a042a45653723ac538b8bb9136461/80b75c7738d3dcdd4bb16e5436db3a7ae7cacbab

Comment (developit) [reply src/plugins/transform-safari-block-shadowing/index.js]: Fixed!

File: src/plugins/transform-safari-block-shadowing/index.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+        const bindings = t.getOuterBindingIdentifiers(path.node.id);
+        for (const name of Object.keys(bindings)) {
+          let scope = path.scope;
+
+          // ignore parent bindings (note: impossible due to let/const?)
+          if (!scope.hasOwnBinding(name)) continue;
+
+          // this only affects block bindings
+          if (t.isFunction(scope.block) || t.isProgram(scope.block)) continue;
```
Comment (jridgewell) [inline src/plugins/transform-safari-block-shadowing/index.js]: This should only be done once, above the loop.

**Comment by developit** [comment]
Published in `@babel/preset-modules@0.1.3`

**Pull request merged**